Makefile

```diff
@@ -1,58 +1,178 @@
 	target=$$(echo $(subst image-,,$@));\
 	make $$target GOOS=linux;\
 	VERSION=$(VERSION) REGISTRY=$(REGISTRY) BUILD_PLATFORMS=linux/$(GOARCH) hack/docker.sh $$target
 
+###################
+# UI Building     #
+###################
 
+.PHONY: bundle-ui-dashboard
 bundle-ui-dashboard:
 	cd ui && pnpm run dashboard:build
-bin-karmada-dashboard-web:
-	BUILD_PLATFORMS=$(GOOS)/$(GOARCH) hack/build.sh karmada-dashboard-web
-image-karmada-dashboard-web:
-	BUILD_PLATFORMS=linux/$(GOARCH) hack/build.sh karmada-dashboard-web
-	cp -R ui/apps/dashboard/dist _output/bin/linux/$(GOARCH)/dist
-	DOCKER_FILE="build-web.Dockerfile" VERSION=$(VERSION) REGISTRY=$(REGISTRY) BUILD_PLATFORMS=linux/$(GOARCH) hack/docker.sh karmada-dashboard-web
 
+###################
+# Dependencies    #
+###################
+
+# Install Go dependencies
+.PHONY: install-deps
+install-deps:
+	go mod tidy
+	go mod verify
+
+# Install UI dependencies
+.PHONY: install-ui-deps
+install-ui-deps:
+	cd ui && pnpm install
+
+# Install all dependencies
+.PHONY: install
+install: install-deps install-ui-deps
+
+###################
+# Development     #
+###################
+
+# Run both API and Web server
+.PHONY: run
+run:
+ifndef KUBECONFIG
+	$(error KUBECONFIG is required. Please specify the path to karmada kubeconfig)
+endif
+	@echo "Starting API server..."
+	@_output/bin/$(GOOS)/$(GOARCH)/karmada-dashboard-api \
+		--karmada-kubeconfig=$(KUBECONFIG) \
+		--karmada-context=$(KARMADA_CTX) \
+		--kubeconfig=$(KUBECONFIG) \
+		--context=$(HOST_CTX) \
+		--insecure-port=$(API_PORT) \
+		$(if $(filter true,$(SKIP_TLS_VERIFY)),--skip-karmada-apiserver-tls-verify) & echo $$! > .api.pid
+	@echo "Starting Web server..."
+	@cd ui && VITE_API_HOST=$(API_HOST) pnpm run dashboard:dev || (kill `cat .api.pid` && rm .api.pid)
+	@if [ -f .api.pid ]; then kill `cat .api.pid` && rm .api.pid; fi
+
+# Run API server only
+.PHONY: run-api
+run-api: build
+ifndef KUBECONFIG
+	$(error KUBECONFIG is required. Please specify the path to karmada kubeconfig)
+endif
+	_output/bin/$(GOOS)/$(GOARCH)/karmada-dashboard-api \
+		--karmada-kubeconfig=$(KUBECONFIG) \
+		--karmada-context=$(KARMADA_CTX) \
+		--kubeconfig=$(KUBECONFIG) \
+		--context=$(HOST_CTX) \
+		--insecure-port=$(API_PORT) \
+		$(if $(filter true,$(SKIP_TLS_VERIFY)),--skip-karmada-apiserver-tls-verify)
+
+# Run Web server only
+.PHONY: run-web
+run-web: install-ui-deps build
+	cd ui && VITE_API_HOST=$(API_HOST) pnpm run dashboard:dev
+
+# Generate JWT token for dashboard login
+.PHONY: gen-token
+gen-token:
+ifndef KUBECONFIG
+	$(error KUBECONFIG is required. Please specify the path to karmada kubeconfig)
+endif
+	@echo "Switching to karmada-apiserver context..."
+	@kubectl config use-context $(KARMADA_CTX)
+	@echo "Creating service account..."
+	@kubectl apply -f artifacts/dashboard/karmada-dashboard-sa.yaml
+	@echo "Getting JWT token..."
+	@kubectl -n karmada-system get secret -l app=karmada-dashboard -o jsonpath='{.items[0].data.token}' | base64 -d
```

Comment:
em, why use label selector here, maybe use secret name better, like 
```
kubectl -n karmada-system get secret karmada-dashboard-secret -ojsonpath='{.data.token}' | base64 -d
```

apart from that, maybe use go-template more compatible here.
```
kubectl -n karmada-system get secret karmada-dashboard-secret  -o go-template="{{.data.token | base64decode}}"
```

Reply:
okk, follow the instructions in development.md.